app/models/zone.rb

```diff
@@ -209,6 +243,7 @@ def ntp_reload_queue
 
   def check_zone_in_use_on_destroy
     raise _("cannot delete default zone") if name == "default"
+    raise _("cannot delete maintenance zone") if self == miq_region&.maintenance_zone
```

Comment:
I don't think `miq_region` can be nil here. probably another thing that needs to be handled in the spec.

Reply:
code based on non-nil assumptions is evil ;)

Reply:
If there is no region, we should blow up way before here and the entire application will fail amazingly.

Reply:
I prefer rule "be paranoid that there can be always nil regardless how improbable it is". It makes code safer. 
So if you're right and it can't happen in 99,9999999%, why not to make code even safer?

Reply:
Because it's needless complexity.  You can assign the bug to me if the region is nil.

Reply:
It makes the code less likely to raise exceptions, but it can make things harder to figure out when something does go wrong.  I'd rather get a bug that says "undefined method maintenance_zone on nil:NilClass` so it is obvious what is wrong.

I agree that ideally we'd have a not null constraint on the DB or model for things like this but if miq_region is nil we really do have bigger problems.

Reply:
I'm going to also say to remove the nil in this particular case.  In addition to what others have said above, another reason is that since there is already a global assumption of a non-nil miq_region, then a reader of this code will read it as "in this particular case, miq_region could be nil", which is inaccurate and confusing.

Reply:
I will probably never understand why we want to live with "There is a global assumption". 
Maybe in this case it is really improbable, but designing big systems with global assumptions of this type doesn't seem as good idea to me.
But if you all wanted it here, then ok.